lib/renderer.coffee

```diff
@@ -284,3 +284,27 @@ module.exports =
           return cb null, e.message
       exts: /\.(yaml)$/i
       lang: -> 'json'
+    'glsl':
+      render: (text, filepath, cb) ->
+        try
+          # glslangValidator needs file not text, so we create temp file.
+          # extension must be the same, so validator can detect shader type.
+          tmp = require 'tmp'
+          extension = '.' + filepath.split('.').pop();
+          tempFile = tmp.fileSync {prefix: 'atom-preview-editor', postfix: extension};
+          file = fs.writeFileSync tempFile.fd, text
```

Comment:
Would be sweet if you could use `file`/`writeFile` instead, so as to not block the thread. Thanks to `cb`, all should still work as expected.